### PR TITLE
Profile page UI - getProfile, updateProfile backend

### PR DESCRIPTION
**Pull request by minhphan156**

### CODE REVIEW

File: back-end/src/routes/api.js [hunk 1]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+        ExpressionAttributeValues:{
+          ":e":email,
+          ":p":password
+        },
+        ReturnValues:"UPDATED_NEW"
+      };
+      try {
+        dynamodb.update(updateProfileParams, (err, data) => {
+          if (err){
+            console.debug("Unable to update item. Error JSON:", JSON.stringify(err, null, 2));
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: change this to 
```js
console.error("...")
```
and remove **all** `console.debug` statements

File: back-end/src/routes/api.js [hunk 2]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+          }
+        });
+      }
+      catch(e) {
+        console.debug(e);
+        response.status(statusCode).send(message);
+      }
+    }
+  })().catch(e => {
+    console.debug(e);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: same thing

File: back-end/src/routes/api.js [hunk 3]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+          try{
+            var photoData = await s3.getObject(getParams).promise();
+            responseObject = {
+              avatar: photoData.Body,
+              email: profileData.email,
+              password: profileData.password,
+              userid: profileData.userid
+            };  
+          } catch(e) {
+            console.debug("can not find the photo", e);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: error

File: back-end/src/routes/api.js [hunk 4]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+      catch(e) {
+        console.error("Could not retrieve information", e);
+      }
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+
+  })().catch(e => {
+    console.debug(e);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: error

File: back-end/src/routes/api.js [hunk 5]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
   });
 };
 
+const updateProfile = (request,response) => {
+  (async() => {
+    var statusCode = 400;
+    var message = "sever error";
+    const email = request.body.email;
```
Comment (evanugarte) [inline back-end/src/routes/api.js]:
```js
const { email, password, userid } = request.body;
```

File: back-end/src/routes/api.js [hunk 6]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+          ":p":password,
+          ":a":avatarName
+        },
+        ReturnValues:"UPDATED_NEW"
+      };
+      try {
+        const uploadAvatar = await s3.upload(uploadParams).promise();
+        dynamodb.update(updateProfileParams, (err, data) => {
+          if (err){
+            console.debug("Unable to update item. Error JSON:", JSON.stringify(err, null, 2));
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: console.error

File: back-end/src/routes/api.js [hunk 7]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+            response.status(statusCode).send(message);
+
+          } else {
+            console.debug("UpdateItem succeeded:", JSON.stringify(data, null, 2));
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: remove all debugs, i learned that from [here](https://medium.com/@hfally/deactivate-console-log-on-production-why-and-how-d0345abbe71)

File: back-end/src/routes/api.js [hunk 8]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+            statusCode = 200;
+            message = "update profile successful";
+            response.status(statusCode).send(message);
+          }
+        });
+       
+      }
+      catch(e) {
+        console.debug(e);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: error

File: back-end/src/routes/api.js [hunk 9]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+        response.status(statusCode).send(message);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: we should do
```js
// for better desctiption
response.status(statusCode).send(e);
```

File: back-end/src/routes/api.js [hunk 10]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+          ":e":email,
+          ":p":password
+        },
+        ReturnValues:"UPDATED_NEW"
+      };
+      try {
+        dynamodb.update(updateProfileParams, (err, data) => {
+          if (err){
+            console.debug("Unable to update item. Error JSON:", JSON.stringify(err, null, 2));
+            response.status(statusCode).send(message);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: we should do
```js
// for better desctiption
response.status(statusCode).send(e);
```

File: back-end/src/routes/api.js [hunk 11]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+            console.debug("UpdateItem succeeded:", JSON.stringify(data, null, 2));
+            statusCode = 200;
+            message = "update profile successful";
+            response.status(statusCode).send(message);
+          }
+        });
+      }
+      catch(e) {
+        console.debug(e);
+        response.status(statusCode).send(message);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: we should do
```js
// for better desctiption
response.status(statusCode).send(e);
```

File: back-end/src/routes/api.js [hunk 12]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+          } else {
+            console.debug("UpdateItem succeeded:", JSON.stringify(data, null, 2));
+            statusCode = 200;
+            message = "update profile successful";
+            response.status(statusCode).send(message);
+          }
+        });
+      }
+      catch(e) {
+        console.debug(e);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: .error

File: back-end/src/routes/api.js [hunk 13]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+        response.status(statusCode).send(message);
+      }
+    }
+  })().catch(e => {
+    console.debug(e);
+    response.status(statusCode).send(message);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: we should do
```js
// for better desctiption
response.status(statusCode).send(e);
```

File: back-end/src/routes/api.js [hunk 14]
```diff
@@ -164,6 +164,155 @@ const getPhotoSocial = (request, response) => {
...
+        console.error("Could not retrieve information", e);
+      }
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+
+  })().catch(e => {
+    console.debug(e);
+    response.status(statusCode).send(message);
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: we should do
```js
// for better desctiption
response.status(statusCode).send(e);
```

File: src/components/ProfilePage.js [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+    };
+    this.setState({ isSubmit: true });
+    this.props.updateProfileFileAction(profileData, this.props.history);
+  }
+
+  handleClickShowPassword = () => {
+    this.setState({showPassword: !this.state.showPassword });
+  };
+
+  handleMouseDownPassword = event => {
```
Comment (evanugarte) [inline src/components/ProfilePage.js]: put () around `event`

File: src/components/ProfilePage.js [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+          >
+            Save Changes
+          </Button>
+        </form>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
```
Comment (evanugarte) [inline src/components/ProfilePage.js]: put () around `state`

File: src/components/SocialPage.js [hunk 17]
```diff
@@ -13,6 +13,8 @@ import FavoriteIcon from "@material-ui/icons/Favorite";
 import ShareIcon from "@material-ui/icons/Share";
 import MoreVertIcon from "@material-ui/icons/MoreVert";
 import { withStyles } from "@material-ui/core/styles";
+import { getProfileFileAction } from "../actions/updateProfileAction";
+import { connect } from "react-redux";
 
 const styles = theme => ({
```
Comment (evanugarte) [inline src/components/SocialPage.js]: () around `theme`

**Comment by evanugarte** [comment]
Looks good, take a look at suggestions

File: back-end/src/routes/api.js (hunk 1 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 2 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 3 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 4 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 5 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 13 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: back-end/src/routes/api.js (hunk 14 above)
Comment (minhphan156) [reply back-end/src/routes/api.js]: fixed

File: src/components/ProfilePage.js (hunk 15 above)
Comment (minhphan156) [reply src/components/ProfilePage.js]: fixed

File: src/components/ProfilePage.js (hunk 16 above)
Comment (minhphan156) [reply src/components/ProfilePage.js]: fixed

File: src/components/SocialPage.js (hunk 17 above)
Comment (minhphan156) [reply src/components/SocialPage.js]: fixed

File: back-end/src/routes/api.js (hunk 7 above)
Comment (evanugarte) [reply back-end/src/routes/api.js]: remove please

File: back-end/src/routes/api.js [hunk 18]
```diff
@@ -164,6 +164,157 @@ const getPhotoSocial = (request, response) => {
...
+        ReturnValues:"UPDATED_NEW"
+      };
+      try {
+        dynamodb.update(updateProfileParams, (err, data) => {
+          if (err){
+            console.error("Unable to update item. Error JSON:", JSON.stringify(err, null, 2));
+            response.status(statusCode).send(err);
+
+          } else {
+            console.debug("UpdateItem succeeded:", JSON.stringify(data, null, 2));
```
Comment (evanugarte) [inline back-end/src/routes/api.js]: remove debug

File: src/actions/updateProfileAction.js [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+  axios
+    .post(" http://localhost:4000/users/updateprofile", formData)
+    .then(res => {
+      dispatch({
+        type: UPDATE_PROFILE,
+        payload: res.data
+      });
+      history.push("/");      
+    })
+    .catch(err => console.debug(err));
```
Comment (evanugarte) [inline src/actions/updateProfileAction.js]: this should be console.error

File: src/actions/updateProfileAction.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+export const getProfileFileAction = (query, history) => dispatch => {
+  axios
+    .post(" http://localhost:4000/users/getprofile", query)
+    .then(res => {
+      dispatch({
+        type: GET_PROFILE,
+        payload: res.data
+      });
+    })
+    .catch(err => console.debug(err));
```
Comment (evanugarte) [inline src/actions/updateProfileAction.js]: this should be console.error

**Pull request merged**